ogr/ogrsf_frmts/miramon/mm_wrlayr.c

```diff
@@ -0,0 +1,7038 @@
+                                                 nNumPrivateMMField,
+                                             hMMFeature->pRecords[nIRecord]
+                                                 .pField[nIField]
+                                                 .pDinValue,
+                                             FALSE))
+                    return 1;
+            }
+            else if (pBD_XP->pField[nIField + nNumPrivateMMField].FieldType ==
+                     'N')
+            {
+                if (pBD_XP->pField[nIField + nNumPrivateMMField].Is64)
+                {
+                    if (MMWriteValueToRecordDBXP(
+                            hMiraMonLayer, pszRecordOnCourse,
+                            pBD_XP->pField + nIField + nNumPrivateMMField,
+                            &hMMFeature->pRecords[nIRecord]
+                                 .pField[nIField]
+                                 .iValue,
+                            TRUE))
+                        return 1;
+                }
+                else
+                {
+                    if (MMWriteValueToRecordDBXP(
+                            hMiraMonLayer, pszRecordOnCourse,
+                            pBD_XP->pField + nIField + nNumPrivateMMField,
+                            &hMMFeature->pRecords[nIRecord]
+                                 .pField[nIField]
+                                 .dValue,
+                            FALSE))
+                        return 1;
+                }
+            }
+            else if (pBD_XP->pField[nIField + nNumPrivateMMField].FieldType ==
+                     'D')
+            {
+                if (MMWriteValueToRecordDBXP(hMiraMonLayer, pszRecordOnCourse,
+                                             pBD_XP->pField + nIField +
+                                                 nNumPrivateMMField,
+                                             hMMFeature->pRecords[nIRecord]
+                                                 .pField[nIField]
+                                                 .pDinValue,
+                                             FALSE))
+                    return 1;
+            }
+        }
+
+        if (MMAppendBlockToBuffer(pFlushRecList))
+            return 1;
+
+        (*nNumRecords)++;
+    }
+    return 0;
+}
+
+// Adds feature records to a MiraMon database associated with a vector layer.
+static int MMDetectAndFixDBFWidthChange(
+    struct MiraMonVectLayerInfo *hMiraMonLayer,
+    struct MiraMonFeature *hMMFeature, struct MMAdmDatabase *pMMAdmDB,
+    struct MM_FLUSH_INFO *pFlushRecList, MM_EXT_DBF_N_FIELDS nNumPrivateMMField,
+    MM_EXT_DBF_N_MULTIPLE_RECORDS nIRecord, MM_EXT_DBF_N_FIELDS nIField)
+{
+    if (!hMiraMonLayer)
+        return 1;
+
+    if (!hMMFeature)
+        return 0;
+
+    if (nIRecord >= hMMFeature->nNumMRecords)
+        return 0;
+
+    if (nIField >= hMMFeature->pRecords[nIRecord].nNumField)
+        return 0;
```

Comment:
shouldn't the return code be 1 in those 3 cases?

Reply:
It not bad if the code gets into here in these conditions but to make it cleared I avoid getting into here if there's nothing to do.